etcdserver: clean up start and stop logic of raft

yichengq opened the pull request:
kill TODO and make it more readable.

follows #3077

xiang90 commented on etcdserver/raft.go:
```diff
@@ -176,6 +184,11 @@ func (r *raftNode) apply() chan apply {
+	<-r.done
+}
+
+func (r *raftNode) close() {
```
can we rename this to shutdown or something else with a comment? this confuses me with `stop`.

xiang90 commented on etcdserver/raft.go:
```diff
@@ -108,10 +108,18 @@ type raftNode struct {
 	done    chan struct{}
 }
 
+func (r *raftNode) start(s *EtcdServer) {
```
can we add a comment about start? also is it possible to combine start and run? Ideally we should get rid of the passed in server structure.

yichengq commented:
Addressed the comments.

> Ideally we should get rid of the passed in server structure.

I tried, then it needs some line `srv.r.s = srv`, which is not easy to read then. It could be wrapped in some member function to be solved though.

xiang90 commented:
> I tried, then it needs some line srv.r.s = srv, which is not easy to read then. It could be wrapped in some member function to be solved though.

I mean `Ideally we should get rid of the passed in server structure`. I do not quite understand what do you mean by srv.r.s = srv. If we get rid of server structure, then there is no s in r.

yichengq commented:
@xiang90 Agree with your point. I have added a TODO for getting rid of server structure.

xiang90 commented on etcdserver/raft.go:
```diff
@@ -176,6 +186,11 @@ func (r *raftNode) apply() chan apply {
+	<-r.done
+}
+
+func (r *raftNode) shutdown() {
```
onStop?

yichengq replied:
changed.

xiang90 commented:
LGTM.
